packages/affiliates/libs/affiliates.js

```diff
@@ -0,0 +1,192 @@
+const { EmpBalancesHistory } = require("./processors");
+const { Prices } = require("./models");
+const { DecodeLog } = require("./contracts");
+const highland = require("highland");
+const moment = require("moment");
+const Promise = require("bluebird");
+const assert = require("assert");
+const { parseFixed } = require("@uma/common");
+
+const DeployerRewards = ({ queries, empCreatorAbi, empAbi, coingecko }) => {
+  assert(coingecko, "requires coingecko api");
+  assert(queries, "requires queries class");
+  assert(empCreatorAbi, "requires creator abi");
+  assert(empAbi, "requires empAbi");
+  async function getBalanceHistory(address, start, end) {
+    // stream is a bit more optimal than waiting for entire query to return as array
+    const stream = await queries.streamLogsByContract(address, start, end);
+    const decode = DecodeLog(empAbi);
+    const balancesHistory = EmpBalancesHistory();
+    await highland(stream)
+      .map(log => {
+        return decode(log, {
+          blockNumber: log.block_number,
+          blockTimestamp: moment(log.block_timestamp.value).valueOf()
+        });
+      })
+      .doto(log => balancesHistory.handleEvent(log.blockNumber, log))
+      .last()
+      .toPromise(Promise);
+
+    return balancesHistory;
+  }
+  async function getAllBalanceHistories(addresses = [], start, end) {
+    return Promise.reduce(
+      addresses,
+      async (result, address) => {
+        result.push([address, await getBalanceHistory(address, start, end)]);
+        return result;
+      },
+      []
+    );
+  }
+  async function getPriceHistory(address, start, currency = "usd") {
```

Comment:
It'd be more readable if you put comments above each of these functions explaining their high level purpose, inputs and outputs.

Reply:
i agree im going to add a task for improving all documentation